core/shared/src/main/scala/cats/effect/IO.scala

```diff
@@ -301,6 +301,11 @@ object IO extends IOCompanionPlatform with IOLowPriorityImplicits {
 
   // utilities
 
+  private[this] val _stub: IO[Nothing] = raiseError(
+    new NotImplementedError("This IO is not implemented"))
+
+  def stub: IO[Nothing] = _stub
```

Comment:
Just realized something interesting: this is going to throw a constant exception with a useless stack trace. We probably should do something like this:

```suggestion
  def stub: IO[Nothing] = {
    val t = new NotImplementedError("this IO is not implemented"))
    raiseError(t)
  }
```

Reply:
yeah, I guess this isn't meant to be used in production code so we should rather focus on "experience" things like stack traces than perf.

btw. why the change to lowercase `this`? :D

Reply:
> btw. why the change to lowercase this? :D

IIRC, in most cases, exceptions have lower-case messages